catalog-connectors/airflow/airflow-package-catalog-connector/airflow_package_catalog_connector/airflow_package_catalog_connector.py

```diff
@@ -0,0 +1,245 @@
+                                scripts_with_operator_class.append(classes_to_analyze[class_name]['file'])
+                            # remove class from todo list
+                            del classes_to_analyze[class_name]
+                            # A new class was discovered that implements
+                            # BaseOperator. Analysis needs to be repeated because
+                            # OTHER classes might extend THIS class.
+                            analysis_complete = False
+                            break
+
+            # Populate return data structure
+            for script in scripts_with_operator_class:
+                operator_key_list.append({
+                    'airflow_package': airflow_package_name,
+                    'file': script})
+
+            # Dump stats
+            self.log.info(f'Analysis of \'{airflow_package_download_url}\' completed. '
+                          f'Located {operator_class_count} operator classes '
+                          f'in {len(scripts_with_operator_class)} Python scripts.')
+            self.log.debug(f'Operator key list: {operator_key_list}')
+        except Exception as ex:
+            self.log.error('Error retrieving operator list from Airflow package '
+                           f'{airflow_package_download_url}: {ex}')
+
+        return operator_key_list
+
+    def read_catalog_entry(self,
+                           catalog_entry_data: Dict[str, Any],
+                           catalog_metadata: Dict[str, Any]) -> Optional[str]:
+        """
+        Fetch the component that is identified by catalog_entry_data from
+        the downloaded Apache Airflow package.
+
+        :param catalog_entry_data: a dictionary that contains the information needed to read the content
+                                   of the component definition
+        :param catalog_metadata: the metadata associated with the catalog in which this catalog entry is
+                                 stored; in addition to catalog_entry_data, catalog_metadata may also be
+                                 needed to read the component definition for certain types of catalogs
+
+        :returns: the content of the given catalog entry's definition in string form
+        """
+        operator_file_name = catalog_entry_data.get('file')
+
+        if hasattr(self, 'tmp_archive_dir') is False:
+            # Log error and return None
+            self.log.error('Error. Cannot fetch operator definition. The '
+                           ' downloaded Airflow package archive was not found.')
+            return None
+
+        # load component source using the provided key
+        component_source = self.tmp_archive_dir / operator_file_name
+        self.log.debug(f'Reading operator source \'{component_source}\' ...')
+        with open(component_source, 'r') as source:
+            return source.read()
+
+        return None
```

Comment:
NIT: my IDE complains that this last return statement is unreachable -- maybe want to wrap a try around the open?